Build failure: python314Packages.reactivex

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

Home-Assistant recently updated to python3.14: https://github.com/NixOS/nixpkgs/commit/3c1bead43b97f21f607cfc50dc38153bd0f3e5cc

Unfortunately this breaks home-assistant with influxdb enabled due to downstream dependency on python314Packages.reactivex, which [has 314 disabled](https://github.com/NixOS/nixpkgs/blob/eaabf13bc8bcfa8841898c0acfd93a3cbb7474d6/pkgs/development/python-modules/reactivex/default.nix#L18) with a link to [this issue](https://github.com/ReactiveX/RxPY/issues/737).

Reactivex now [lists compatibility with 314](https://github.com/ReactiveX/RxPY/blob/d0b472b0de7771ac3182bda52c01373143af89db/pyproject.toml#L20) and I'm able to at least install it imperatively on 314, but I'm not yet familiar with the package to test its functionality.

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/320081869

### Relevant log output

```console
… while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:37:12:
           36|
           37|   strict = derivationStrict drvAttrs;
             |            ^
           38|

       … while evaluating derivation 'nixos-system-homeslice-26.05.20260405.68d8aa3'
         whose name attribute is located at /nix/store/gay80fqbpm2wakbsyd4in44gx0cwx3h5-source/pkgs/stdenv/generic/make-derivation.nix:541:11

       … while evaluating attribute 'buildCommand' of derivation 'nixos-system-homeslice-26.05.20260405.68d8aa3'
         at /nix/store/gay80fqbpm2wakbsyd4in44gx0cwx3h5-source/nixos/modules/system/activation/top-level.nix:64:7:
           63|       passAsFile = [ "extraDependencies" ];
           64|       buildCommand = systemBuilder;
             |       ^
           65|

       … while evaluating the option `environment.etc."systemd/system".source':

       … while evaluating the option `systemd.units."home-assistant.service".text':

       … while evaluating definitions from `/nix/store/gay80fqbpm2wakbsyd4in44gx0cwx3h5-source/nixos/modules/system/boot/systemd.nix':

       … while evaluating the option `systemd.services.home-assistant.environment':

       … while evaluating definitions from `/nix/store/gay80fqbpm2wakbsyd4in44gx0cwx3h5-source/nixos/modules/services/home-automation/home-assistant.nix':

       … while calculating requiredPythonModules for python3.14-influxdb-client-1.50.0:

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: reactivex-4.1.0 not supported for interpreter python3.14
```

### Additional context

Setting `disabled = false` results in a fair number of test failures:

```
error: Cannot build '/nix/store/hnvq6k4wnp0zci7xvwf1p4540pjb2ay0-python3.14-reactivex-4.1.0.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/1hynbwjyb0fyqhivhwmp7j99rws4qnrr-python3.14-reactivex-4.1.0-dist
         /nix/store/rmjxryvr606wdrr5r538cbmp4ih86a30-python3.14-reactivex-4.1.0
       Last 25 log lines:
       > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > FAILED tests/test_observable/test_flatmap_async.py::TestFlatMapAsync::test_flat_map_async - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_observable/test_fromfuture.py::TestFromFuture::test_future_cancel - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_observable/test_fromfuture.py::TestFromFuture::test_future_dispose - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_observable/test_fromfuture.py::TestFromFuture::test_future_failure - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_observable/test_fromfuture.py::TestFromFuture::test_future_success - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_observable/test_start.py::TestStart::test_start_async - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_observable/test_start.py::TestStart::test_start_async_error - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_observable/test_tofuture.py::TestToFuture::test_await_empty_observable - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_observable/test_tofuture.py::TestToFuture::test_await_error - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_observable/test_tofuture.py::TestToFuture::test_await_success - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_observable/test_tofuture.py::TestToFuture::test_await_success_on_sequence - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_observable/test_tofuture.py::TestToFuture::test_await_with_delay - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_observable/test_tofuture.py::TestToFuture::test_cancel - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_observable/test_tofuture.py::TestToFuture::test_dispose_on_cancel - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_scheduler/test_eventloop/test_asyncioscheduler.py::TestAsyncIOScheduler::test_asyncio_schedule_action - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_scheduler/test_eventloop/test_asyncioscheduler.py::TestAsyncIOScheduler::test_asyncio_schedule_action_cancel - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_scheduler/test_eventloop/test_asyncioscheduler.py::TestAsyncIOScheduler::test_asyncio_schedule_action_due - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_scheduler/test_eventloop/test_asyncioscheduler.py::TestAsyncIOScheduler::test_asyncio_schedule_now - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_scheduler/test_eventloop/test_asynciothreadsafescheduler.py::TestAsyncIOThreadSafeScheduler::test_asyncio_threadsafe_schedule_action - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_scheduler/test_eventloop/test_asynciothreadsafescheduler.py::TestAsyncIOThreadSafeScheduler::test_asyncio_threadsafe_schedule_action_cancel - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_scheduler/test_eventloop/test_asynciothreadsafescheduler.py::TestAsyncIOThreadSafeScheduler::test_asyncio_threadsafe_schedule_action_due - RuntimeError: There is no current event loop in thread 'MainThread'.
       > FAILED tests/test_scheduler/test_eventloop/test_asynciothreadsafescheduler.py::TestAsyncIOThreadSafeScheduler::test_asyncio_threadsafe_schedule_now - RuntimeError: There is no current event loop in thread 'MainThread'.
       > =========== 22 failed, 1319 passed, 10 skipped, 3 warnings in 19.53s ===========
       For full logs, run:
         nix log /nix/store/hnvq6k4wnp0zci7xvwf1p4540pjb2ay0-python3.14-reactivex-4.1.0.drv
error: Cannot build '/nix/store/4bwz9k3imllgyh3pnfh7fkyllrxs5syj-python3-3.14.3-env.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/0grpzykip6k3y4k6gc374f1rcid8cm5n-python3-3.14.3-env
```

I attempted to override the dependency for home-assistant with a similar approach (below) ~and unfortunately my initial attempt was unsuccessful (still gives me the "disabled" error)~. EDIT: was overcomplicating things, hass takes a `packageOverrides` argument directly so this is my workaround for the moment:

```nix
home-assistant.package = pkgs.home-assistant.override {
  packageOverrides = _: super: {
    reactivex = super.reactivex.overridePythonAttrs {
      disabled = false;
      doCheck = false;
    };
  };
};
```

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.68, NixOS, 26.05 (Yarara), 26.05.20260318.b40629e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.3`
 - channels(root): `""`
 - nixpkgs: `/nix/store/ap9dpkyzikzzh04259wlsvha2mw455x4-source`

### Notify maintainers

@fabaff 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/reactivex/default.nix
+++ b/pkgs/development/python-modules/reactivex/default.nix
@@ -2,10 +2,10 @@
   lib,
   buildPythonPackage,
   fetchFromGitHub,
+  fetchpatch,
   poetry-core,
   pytest-asyncio,
   pytestCheckHook,
-  pythonAtLeast,
   typing-extensions,
 }:
 
@@ -14,15 +14,21 @@ buildPythonPackage rec {
   version = "4.1.0";
   pyproject = true;
 
-  # https://github.com/ReactiveX/RxPY/issues/737
-  disabled = pythonAtLeast "3.14";
-
   src = fetchFromGitHub {
     owner = "ReactiveX";
     repo = "RxPY";
     tag = "v${version}";
     hash = "sha256-napPfp72gqy43UmkPu1/erhjmJbZypHZQikmjIFVBqA=";
   };
+
+  patches = [
+    # Upstream PR: https://github.com/ReactiveX/RxPY/pull/728
+    (fetchpatch {
+      name = "python-3.14.patch";
+      url = "https://github.com/ReactiveX/RxPY/commit/78f4a594ca2b0e27ad93ec0e1b1c0d56d5d6540d.patch";
+      hash = "sha256-1GQm/4BTd5ZnIqfEUSb0Ja3w0y1R9EoFpzwua7gpIzo=";
+    })
+  ];
 
   build-system = [ poetry-core ];
 

--- a/pkgs/servers/home-assistant/tests.nix
+++ b/pkgs/servers/home-assistant/tests.nix
@@ -172,6 +172,13 @@ let
       # outdated snapshot
       "tests/components/hypontech/test_sensor.py::test_sensors"
     ];
+    influxdb = [
+      # These tests fail because they check for the number of warnings in the
+      # logs and there is an extra warning in the logs:
+      # `WARNING:aiohttp_fast_zlib:zlib_ng and isal are not available, falling back to zlib, performance will be degraded.`
+      "tests/components/influxdb/test_sensor.py::test_state_for_no_results"
+      "tests/components/influxdb/test_sensor.py::test_state_matches_first_query_result_for_multiple_return"
+    ];
     jellyfin = [
       # AssertionError: assert 'audio/x-flac' == 'audio/flac'
       "tests/components/jellyfin/test_media_source.py::test_resolve"

